### PR TITLE
[docs] Clarify optional protocol support with class vs interface

### PR DESCRIPTION
**Pull request by chamons**
- From https://github.com/xamarin/ios-samples/pull/410

### CODE REVIEW

**Review by spouliot — COMMENTED** [review]
Those `docs/` files pre-dates the open sourcing / new workflow of the documentation. This might not apply cleanly anymore.

**Comment by chamons** [comment]
Is there a better place to land this @spouliot ?

**Comment by spouliot** [comment]
https://github.com/MicrosoftDocs/xamarin-docs (public) or https://github.com/MicrosoftDocs/xamarin-docs-pr (private, for internal  pull-request).

File: docs/website/binding_types_reference_guide.md [hunk 1]
```diff
@@ -879,8 +879,37 @@ class MyDelegate : NSObject, IUITableViewDelegate {
...
+```
+
+There is a slight behavior difference between the two at runtime.
+
+- Users of the base class (NSUrlSessionDownloadDelegate in example) provides all
+required and optional selectors, returning reasonable default values.
+- Users of the interface (INSUrlSessionDownloadDelegate in example) only respond
+to the exact selectors provided.
+
+Some rare classes can beheave differently here. In almost all cases however
```
Comment (tj-devel709) [inline docs/website/binding_types_reference_guide.md:910]:
```suggestion
Some rare classes can behave differently here. In almost all cases however
```

File: docs/website/binding_types_reference_guide.md [hunk 2]
```diff
@@ -879,8 +879,37 @@ class MyDelegate : NSObject, IUITableViewDelegate {
 }
 ```
 
-It does not matter if the interface is implemented
-implicitly or explicitly.
+This is work for all required protocol members, but there
```
Comment (tj-devel709) [inline docs/website/binding_types_reference_guide.md:882]:
```suggestion
This will work for all required protocol members, but there
```

**Review by tj-devel709 — APPROVED** [review]
Added a few word change suggestions, but other than that it all looks good to me!

**Comment by chamons** [comment]
@spouliot also done in https://github.com/MicrosoftDocs/xamarin-docs-pr/pull/1891

Should I land this to keep it in sync, or nuke the entire dir in another PR?

**Review by whitneyschmidt — APPROVED** [review]

**Comment by spouliot** [comment]
@chamons better do a diff before nuking
in case some stuff sneaking here and not there...

**Review by mandel-macaque — APPROVED** [review]

**Review by rolfbjarne — APPROVED** [review]

**Review by dalexsoto — APPROVED** [review]

**Pull request merged**